flocker/node/functional/test_deploy.py

```diff
@@ -47,30 +47,37 @@ def __init__(self, hostname, volume_service, docker_client=None,
     def discover_state(self, local_state):
         d = self.manifestations_deployer.discover_state(local_state)
 
-        def got_manifestations_state(manifestations_state):
-            manifestations_state = manifestations_state[0]
+        def got_manifestations_state(manifestations_local_state):
+            manifestations_state = (
+                manifestations_local_state.shared_state_changes()[0])
```

Comment:
Given `manifestations_local_state` has an explicit attribute with the object we want, seems like that's better than using a hard to understand and brittle index into a tuple.